# Migration 0011 - Заполнение полей `error_text` при создании
# Generated by Django 5.1.6 on 2025-04-05 11:41

from django.db import migrations

def set_error_text_values(apps, schema_editor):
    Field = apps.get_model('backend', 'Field')
    error_texts = {
        "username": "Имя пользователя должно быть уникальным, не должно содержать пробелов и быть от 4 до 16 символов.",
        "password": "Пароль должен быть длиннее 8 символов и содержать хотя бы одну цифру.",
        "last_name": None,
        "first_name": None,
        "patronymic": None,
    }

    for field_name, error_text in error_texts.items():
        field = Field.objects.get(key_name=field_name)
        field.error_text = error_text
        field.save()

class Migration(migrations.Migration):

    # ЗДЕСЬ ОН МИГРАЦИЯ 0011
    dependencies = [
        ('backend', '0010_field_error_text'),
    ]

    operations = [
        migrations.RunPython(set_error_text_values),
    ]